message/message_matcher_parser.y

```diff
@@ -130,12 +129,30 @@ string_test : string_vars relational STRING_VALUE
        //fmt.Println("string_test regexp", $1, $2, $3)
        nodes = append(nodes, &tree{stmt:&Statement{$1, $2, $3}})
        }
+   |   VAR_TIMESTAMP relational STRING_VALUE
+       {
+       //fmt.Println("timestamp_string_test regexp", $1, $2, $3)
+       timestamp, err := parseFullDateTime($3.token)
+
+       if err != nil {
+            log.Printf("Invalid datetime value for Timestamp field: %s", err)
+            return 1
+       }
+       $3.tokenId = NUMERIC_VALUE
+       $3.double = float64(timestamp.UnixNano())
+       nodes = append(nodes, &tree{stmt:&Statement{$1, $2, $3}})
+       }
 ;
 numeric_test : numeric_vars relational NUMERIC_VALUE
-   {
-   //fmt.Println("numeric_test", $1, $2, $3)
-   nodes = append(nodes, &tree{stmt:&Statement{$1, $2, $3}})
-   }
+        {
+        //fmt.Println("numeric_test", $1, $2, $3)
+        nodes = append(nodes, &tree{stmt:&Statement{$1, $2, $3}})
+         }
+   |    VAR_TIMESTAMP relational NUMERIC_VALUE
```

Comment:
You should be able to leave VAR_TIMESTAMP in numeric_vars and get rid of this grammar (the string case is special cased and tested first)